src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -184,11 +186,21 @@ public FileUploadSubmission save(FileUploadSubmission fileUploadSubmission, Mult
         if (exerciseDueDate != null && exerciseDueDate.isBefore(ZonedDateTime.now()) && participation.getInitializationDate().isBefore(exerciseDueDate)) {
             throw new ResponseStatusException(HttpStatus.FORBIDDEN);
         }
+        if (file.isEmpty()) {
+            throw new IOException("Failed to store empty file" + file.getName());
+        }
+        var multipartFileHash = DigestUtils.md5Hex(file.getInputStream());
         // check if we already had file associated with this submission
         fileUploadSubmission.onDelete();
```

Comment:
Yes this will leave the user with no files, which we should not do. I tried to simply move it, but the problem is that `onDelete` will remove the latest file stored in the folder of this submission and in this case it is newly uploaded file. I will check how can I do this in a different way.

Reply:
I was able to fix this. Now we don't delete the directory were files are stored, but only the old file itself. So as you suggested the old file is deleted only after we have checked the new file is stored properly.